QuizApp/QuizzesViewController.swift

```diff
@@ -93,16 +95,21 @@ class QuizzesViewController: UIViewController {
     @objc
     func customAction() {
         tableView.isHidden = false
-        quizzes = DataService().fetchQuizes()
-        self.quizzesGroupedByCategory = groupByCategory(quizzesList: quizzes)
-        tableView.reloadData()
-        
-        let count = quizzes.map{$0.questions}.flatMap{$0}.filter{$0.question.contains("NBA")}.count
-        
-        fact.text = "Fun Fact"
-        nbaQuestion.text = "There are \(count) questions that contain the word \"NBA\""
+        quizzesUseCase.fetchQuizzes()  { quizzes in
```

Comment:
There is a strong reference cycle happening here. Self has a reference to quizzesUseCase and then quizzesUseCase has a reference to self inside the closure. Use `[weak self]` to break the cycle.